Content improvements

kaishin opened the pull request:
This PR aims to introduce the content changes that the SWWG have worked on since last year.
Several changes have been made to the content of the most visited pages of the website, including the landing page, the getting started page, and the install page.

The main changes are:

- New content added to the landing page to give more guidance to visitors via clear navigation buttons, code snippets, and two new "Use cases" and "Getting involved" sections.
- The "Getting started" page has been reworked to provide a more structured and guided approach to using Swift. It now features links to four brand new guides to get started, with additional resources for seasoned users in the "Go further" section.
- The "Install" page has been streamlined for more clarity on how to install Swift on various platforms. Each platform now has its own dedicated page.

The PR can be [previewed here](https://swift-org-website-kaishin.vercel.app/).

## Checklist

The following are steps that need to be taken before merging this PR.

- [ ] Update code snippet placeholders (@TimTr)
- [ ] Publish guide source code examples

amartini51 commented on _data/go_further.yml:
```diff
@@ -0,0 +1,29 @@
+- title: "Value and Reference Types"
+  description: "This article describes the differences in behavior between _value types_ and _reference types_—a fundamental part of learning Swift and choosing between structures and classes."
+  content_type: article
+  content_url: https://www.swift.org/blog/2023-04-29-value-and-reference-types
```
This link returns a 404 error.

tomerd replied:
yes, this is yet to be published. @dempseyatgithub should we remove for now and then add back after the post is published?

tomerd commented on getting-started/cli-swiftpm/index.md:
```diff
@@ -0,0 +1,232 @@
+To start, we’ll use SwiftPM to make a new project for us. In your terminal of choice run:
+
+~~~bash
+❯ mkdir MyCLI
+❯ cd MyCLI
+❯ swift package init --name MyCLI --type executable
+~~~
```
https://github.com/apple/swift-org-website/pull/357

alexandersandberg commented on assets/stylesheets/pages/_landing.scss:
```diff
@@ -0,0 +1,163 @@
+.preamble {
+  font-weight: normal;
```
I think it would be better to use a font weight of `300` instead of `normal` (`400`) for the preamble to increase the visual contrast between the regular and the emphasized words.

**Before (`normal`):**
<img width="779" alt="CleanShot 2023-08-18 at 09 54 46" src="https://github.com/apple/swift-org-website/assets/35671299/17278dd0-f86e-4865-b34d-c5a47d2b8bfc">

**After (`300`):**
<img width="779" alt="CleanShot 2023-08-18 at 09 54 31" src="https://github.com/apple/swift-org-website/assets/35671299/93306aab-743a-4710-b3b1-6c7a3af359c9">

What do we think?

cc: @daveverwer 

```suggestion
```

kaishin replied:
I agree it looks better. While we're revisiting this I'd also increase the line height by .1 or .2.

alexandersandberg replied:
Sounds good. Will fix later. 👍

daveverwer replied:
Looks great 👍

alexandersandberg replied:
https://github.com/apple/swift-org-website/pull/354/commits/ebc14d95d32dcf8bdf20a4b29c048bcffb198a22

alexandersandberg commented on assets/stylesheets/pages/_landing.scss:
```diff
@@ -0,0 +1,163 @@
+          }
+        }
+
+        img {
+          filter: invert(1);
+
+          @media (prefers-color-scheme: dark) {
+            filter: none;
+          }
+        }
+      }
+
+      img {
+        height: 64px;
+        width: 64px;
+        display: block;
+        margin: 0 auto;
+
+        @media (prefers-color-scheme: dark) {
+          filter: invert(1);
+        }
+      }
```
We can't set the color of the SVGs based on `prefers-color-scheme` only as this can be overridden with the theme selector (Light or Dark instead of Auto).

Example: If your device is set to prefer a **dark** color scheme, but you choose the **Light** option on Swift.org, the SVGs will be invisible:

<img width="696" alt="CleanShot 2023-08-18 at 10 11 43" src="https://github.com/apple/swift-org-website/assets/35671299/f1f7cb93-b13a-4028-994d-7d079246611d">


cc: @kaishin

kaishin replied:
Good catch. I'll add a todo.

stevapple commented on install/index.md:
For Windows toolchains we may have different installation steps for each releases, and for Linux we may have different dependencies. I wonder if these information can be kept and queried for previous releases?

alexandersandberg replied:
cc: @shahmishal

shahmishal replied:
https://github.com/apple/swift-org-website/pull/382

stevapple commented on install/windows/index.md:
The Scoop option is dropped. Intentionally?

alexandersandberg replied:
cc: @shahmishal

xwu commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+<div class="links links-list-nostyle" markdown="1">
+  - [Visit the Swift.org blog](/blog/)
+  - [Visit the Swift forums](https://forums.swift.org)
+  - [Follow @Swiftlang on Twitter](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```
```suggestion
  - [Follow @Swiftlang on X](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```

cthielen commented on install/linux/_docker.md:
```diff
@@ -0,0 +1,32 @@
+
+#### Using Docker Images
+
+0. Pull the Docker image from [Docker hub](https://hub.docker.com/_/swift/):
```
"Docker Hub", per their website

tomerd replied:
```suggestion
0. Pull the Docker image from [Docker Hub](https://hub.docker.com/_/swift/):
```

alexandersandberg commented on assets/images/landing-page/book.svg:
I agree. Visually, they are quite heavy at the moment.

cc: @kaishin

alexandersandberg commented on index.md:
````diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+```swift
+{{ snippet -}}
+```
+{: class="featured-snippet {% if forloop.first %}visible{% endif %}" }
````
Currently, the height is fixed to not cause visual shifts on refresh. The current code snippets are also just quick examples that haven't been adjusted to take the space into account, which means that the gray space issue is more apparent.

Once we have the actual snippets in place, we should revisit this though to see if we need to adjust anything.

I like the idea of a shuffle button too. 👍

tomerd commented on getting-started/_use-cases.md:
```diff
@@ -0,0 +1,62 @@
+  <li class="use-case">
+    <h3>Command-line Tool</h3>
+    <p class="description">
+      Learn how to create cross-platform command-line tool such as short-lived interactive programs, or long-lived daemon that runs in the background.
```
```suggestion
      Learn how to create cross-platform command-line tools such as short-lived interactive programs, or long-lived daemons that run in the background.
```

tomerd commented on about/index.md:
```diff
@@ -3,15 +3,14 @@ layout: page
 ---
 
-Swift is a general-purpose programming language built using a modern approach to safety, performance, and software design patterns.
+Swift is a general-purpose programming language that's approachable for newcomers and powerful in expert hands. It is quick to write, fast to run, modern, and safe.
```
@tkremenek @airspeedswift opinions? the "quick to write, fast to run" is nice play IMO

cthielen replied:
Let's match the front-page with: "Swift is a general-purpose programming language that's approachable for newcomers and powerful for experts. It is fast, modern, safe, and a joy to write."

tomerd commented on about/index.md:
```diff
@@ -3,15 +3,14 @@ layout: page
-Swift is a general-purpose programming language built using a modern approach to safety, performance, and software design patterns.
+Swift is a general-purpose programming language that's approachable for newcomers and powerful in expert hands. It is quick to write, fast to run, modern, and safe.
 
-The goal of the Swift project is to create the best available language for uses ranging from systems programming, to mobile and desktop apps, scaling up to cloud services.  Most importantly, Swift is designed to make writing and maintaining *correct* programs easier for the developer. To achieve this goal, we believe that the most obvious way to write Swift code must also be:
```
@cthielen this whole paragraph was removed, or am I misunderstanding the comment?

tomerd commented on about/index.md:
```diff
@@ -3,15 +3,14 @@ layout: page
-The goal of the Swift project is to create the best available language for uses ranging from systems programming, to mobile and desktop apps, scaling up to cloud services.  Most importantly, Swift is designed to make writing and maintaining *correct* programs easier for the developer. To achieve this goal, we believe that the most obvious way to write Swift code must also be:
+* Swift is general-purpose and modern. Suitable for everything from systems programming, through mobile and desktop apps, to cloud services.
+* Swift is safe. Undefined behavior is the enemy of safety, and it’s best to catch mistakes before software goes into production. Swift makes the obvious path the safest.
+* Swift is fast to run and quick to write. Swift is fast to run and quick to write. It comes with predictable and consistent performance that is on-par with C-based languages without sacrificing developer friendliness.
```
```suggestion
* Swift is fast to run and quick to write. It comes with predictable and consistent performance that is on-par with C-based languages without sacrificing developer friendliness.
```

tomerd commented on getting-started/_use-cases.md:
```diff
@@ -0,0 +1,62 @@
+  <li class="use-case">
+    <h3>Library</h3>
+    <p class="description">
+      Learn how to create cross-platform library for sharing reusable code, or modularize large code-base.
```
```suggestion
      Learn how to create a cross-platform library for sharing reusable code, or modularize large code-base.
```

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+    <p>
+      Swift's small memory footprint, quick startup time, and deterministic performance make it a great choice for server and other networked applications.
+      <br><br>
+      SwiftNIO and Swift's dynamic server ecosystem bring joy to developing networked applications.
```
@cthielen the goal here was to put a light on Swift's package ecosystem in addition to the language itself

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+      Writing Swift is interactive and fun, the syntax is concise yet expressive.
+      Swift code is safe by design and produces software that runs lightning-fast.
+      <br><br>
+      SwiftArgumentParser and Swift's growing package ecosystem make developing cross-platform command-line tools a breeze.
```
@cthielen the goal here was to put a light on Swift's package ecosystem in addition to the language itself

tomerd replied:
@tkremenek @airspeedswift opinions?

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+    <a href="/getting-started/cli-swiftpm" class="cta-secondary">Learn more</a>
+  </li>
+  <li>
+    <h3>Server and Networking</h3>
```
this is intentional, networking applications are more than just web services, and server is more than just networking. that said, I dont think we need to treat them as separate use cases as they are related

tomerd replied:
@tkremenek @airspeedswift opinions?

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+  <li>
+    <h3>Cross-platform Command-line</h3>
+    <p>
+      Writing Swift is interactive and fun, the syntax is concise yet expressive.
```
@tkremenek @airspeedswift opinions?

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+  <li>
+    <h3>Apple platforms</h3>
+    <p>
+      Swift is a powerful and intuitive programming language optimized when running on iOS, macOS, and other Apple platforms.
```
@tkremenek @airspeedswift opinions?

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+    <li>
+      <a href="/documentation">
+        <img src="/assets/images/landing-page/book.svg" />
+        Read the docs
```
its a good point, but I am not sure which better. it a link to the documentation area which is not directly "learn swift". also "read the docs" is term of art in the context.

tomerd replied:
@tkremenek @airspeedswift opinions?

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+        <div class="flex-container">
+          <div class="latest-release-container">
+          <span>
+            5.8
```
I was also on the fence, I think we can keep it as is and change later if we need to

tomerd commented on install/index.md:
```diff
@@ -0,0 +1,17 @@
+
+<hr>
+
+{% include_relative macos/_macos.md %}
```
cc @kaishin

alexandersandberg replied:
I improved this in a55db67. For some reason, we're using `padding-top` instead of `margin-top` on article headings, which makes the spacing between an hr and a heading tag not take advantage of the collapsible margins. I didn't want to accidentally break something elsewhere, so I added some special cases instead.

We will revisit this implementation in the later redesign anyway.

This is what it looks like now:

<img width="469" alt="CleanShot 2023-08-27 at 16 12 33" src="https://github.com/apple/swift-org-website/assets/35671299/d06d9111-aa23-4f1a-b5f9-ca8afa2380a1">

tomerd commented on install/index.md:
```diff
@@ -0,0 +1,17 @@
+Swift development is officially supported on Apple platforms, Linux, and Windows.
+More details about [platform support](/platform-support).
+
+### Latest Release: <a href="/download#releases">Swift {{ site.data.builds.swift_releases.last.name }}</a>
```
@shahmishal maybe the link is redundant? I think the information that says which is the latest version is important

alexandersandberg replied:
I removed the link but kept all the text/information. I think this is better. Let me know what you think.

<img width="783" alt="CleanShot 2023-08-27 at 16 15 38" src="https://github.com/apple/swift-org-website/assets/35671299/c7eb3f8a-ae23-4b10-b14d-03d2abf26693">

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+<div class="links links-list-nostyle" markdown="1">
+  - [Visit the Swift.org blog](/blog/)
+  - [Visit the Swift forums](https://forums.swift.org)
+  - [Follow @Swiftlang on Twitter](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```
```suggestion
  - [Follow @swiftlang on X (formally Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```

amartini51 replied:
The rewrite should be "formerly Twitter", not "formally".

tomerd commented on getting-started/_use-cases.md:
```diff
@@ -0,0 +1,62 @@
+  <li class="use-case">
+    <h3>Web Service</h3>
+    <p class="description">
+      Learn how to create a web-service using a web framework.
```
```suggestion
      Learn how to create a web service using a web framework.
```

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+
+## Getting involved
+
+Everyone is welcome to contribute to Swift. Contributing doesn’t just mean submitting pull requests—there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```
```suggestion
Everyone is welcome to contribute to Swift. Contributing doesn’t just mean wiring code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```

amartini51 replied:
I think "wiring" in the rewrite should be "writing".

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+</div>
+
+<div class="links links-list-nostyle" markdown="1">
+  - [Learn more about Swift](/about)
```
@daveverwer @dempseyatgithub opinions?

tomerd replied:
I think we should remove this link

tomerd commented on about/index.md:
```diff
@@ -3,15 +3,14 @@ layout: page
+Swift is a general-purpose programming language that's approachable for newcomers and powerful in expert hands. It is quick to write, fast to run, modern, and safe.
 
-The goal of the Swift project is to create the best available language for uses ranging from systems programming, to mobile and desktop apps, scaling up to cloud services.  Most importantly, Swift is designed to make writing and maintaining *correct* programs easier for the developer. To achieve this goal, we believe that the most obvious way to write Swift code must also be:
+* Swift is general-purpose and modern. Suitable for everything from systems programming, through mobile and desktop apps, to cloud services.
```
@daveverwer @dempseyatgithub thoughts?

daveverwer replied:
Again, apologies for missing being tagged here.

I'd be fine with these being broken apart into two points. 👍

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+
+## Getting involved
+
+Everyone is welcome to contribute to Swift. Contributing doesn’t just mean wiring code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```
```suggestion
Everyone is welcome to contribute to Swift. Contributing doesn’t just mean writing code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+<div class="links links-list-nostyle" markdown="1">
+  - [Visit the Swift.org blog](/blog/)
+  - [Visit the Swift forums](https://forums.swift.org)
+  - [Follow @swiftlang on X (formally Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```
```suggestion
  - [Follow @swiftlang on X (formerly Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```

stevapple commented on install/windows/index.md:
```diff
@@ -61,47 +21,17 @@ Windows has the following additional platform specific dependencies:
 
-   #### With Winget (Windows Package Manager):
-   ~~~ pwsh
+   ~~~ cmd
```
This part is updated in #361, please take a look.

alexandersandberg replied:
I have brought these changes into the PR. Feel free to double-check that the merge conflicts were resolved correctly. 🙂

stevapple commented on install/windows/index.md:
```diff
@@ -181,22 +107,61 @@ copy /Y %SDKROOT%\usr\share\winsdk.modulemap "%UniversalCRTSdkDir%\Include\%UCRT
 Because it is installing the files into the Visual Studio image, the files will need to be copied each time Visual Studio is updated.
 
-##### Repairing after Visual Studio Updates
+#### Repairing after Visual Studio Updates
```
This step is no longer necessary for Swift 5.9+, we can add it to the documentation.

stevapple commented on install/windows/index.md:
```diff
@@ -181,22 +107,61 @@ copy /Y %SDKROOT%\usr\share\winsdk.modulemap "%UniversalCRTSdkDir%\Include\%UCRT
-$ swift --version
-Apple Swift version 2.2-dev (LLVM ..., Clang ..., Swift ...)
-~~~
+   ~~~ cmd
```
Is this really a CMD command? Looks like PowerShell. We should change the styling and guide the user to use the correct shell.

stevapple commented on install/windows/index.md:
```diff
@@ -181,22 +107,61 @@ copy /Y %SDKROOT%\usr\share\winsdk.modulemap "%UniversalCRTSdkDir%\Include\%UCRT
+
+   or:
+
+   ~~~ cmd
```
Same here. This is **definitely** a PowerShell command that won’t work under CMD.

SimplyDanny commented:
Referring to issue #281, I see a clear improvement that's caused by the fact that download and installation instructions are now separated. However, it's still surprising that the expandable "Older Releases", as well as the tables it hides, are indented. It's okay to hide them by default. But why put them on their own level by indenting them. Aren't they on the same hierarchy as the current release?

<img width="778" alt="Screenshot 2023-08-24 at 23 12 33" src="https://github.com/apple/swift-org-website/assets/16365760/5b57e73d-e2b2-4cb6-a1a1-0f1a6bc2db84">

natikgadzhi commented:
Bonus points: this PR closes #333.

alexandersandberg commented on _data/builds/swift_releases.yml:
```diff
@@ -1301,4 +1301,40 @@
-        - x86_64
\ No newline at end of file
+        - x86_64
+- name: 5.8.1
+  tag: swift-5.8.1-RELEASE
+  xcode: Xcode 14.3.1
+  xcode_release: true
+  date: 2023-06-01
+  platforms:
+    - name: Ubuntu 16.04
+      docker: 5.8.1-xenial
+      archs:
+        - x86_64
+    - name: Ubuntu 18.04
+      docker: 5.8.1-bionic
+      archs:
+        - x86_64
+    - name: Ubuntu 20.04
+      docker: 5.8.1-focal
+      archs:
+        - x86_64
+        - aarch64
+    - name: Ubuntu 22.04
+      docker: 5.8.1-jammy
+      archs:
+        - x86_64
+        - aarch64
+    - name: Amazon Linux 2
+      docker: 5.8.1-amazonlinux2
+      archs:
+        - x86_64
+        - aarch64
+    - name: CentOS 7
+      docker: 5.8.1-centos7
+      archs:
+        - x86_64
+    - name: Windows 10
+      archs:
+        - x86_64
```
@shahmishal Please double-check that this is correct. I wasn't sure about the `xcode_release` flag.

alexandersandberg reviewed (commented):
I brought the PR up-to-date with `main` and fixed some merge conflicts.

alexandersandberg commented:
> However, it's still surprising that the expandable "Older Releases", as well as the tables it hides, are indented. It's okay to hide them by default. But why put them on their own level by indenting them. Aren't they on the same hierarchy as the current release?

@SimplyDanny I agree with you that this should not be part of the latest release section. The indentation is a styling issue though as we apply a 40px left margin to all details tags. But we might even want a dedicated "Older releases" section as simply removing the indentation doesn't really fix the issue.

alexandersandberg commented on _data/builds/swift_releases.yml:
```diff
@@ -0,0 +1,1341 @@
+    - name: Windows 10
+      archs:
+        - x86_64
+- name: 5.8.1
+  tag: swift-5.8.1-RELEASE
+  xcode: Xcode 14.3.1
+  xcode_release: true
+  date: 2023-06-01
+  platforms:
+    - name: Ubuntu 18.04
+      docker: 5.8.1-bionic
+      archs:
+        - x86_64
+    - name: Ubuntu 20.04
+      docker: 5.8.1-focal
+      archs:
+        - x86_64
+        - aarch64
+    - name: Ubuntu 22.04
+      docker: 5.8.1-jammy
+      archs:
+        - x86_64
+        - aarch64
+    - name: CentOS 7
+      docker: 5.8.1-centos7
+      archs:
+        - x86_64
+    - name: Amazon Linux 2
+      docker: 5.8.1-amazonlinux2
+      archs:
+        - x86_64
+        - aarch64
+    - name: Red Hat Universal Base Image 9
+      docker: 5.8.1-rhel-ubi9
+      archs:
+        - x86_64
+        - aarch64
+    - name: Windows 10
+      archs:
+        - x86_64
```
@shahmishal Please double-check that the data that I added for 5.8.1 is correct.

alexandersandberg commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+  - [Learn more about Swift](/about)
+</div>
+
+## Use cases
```
I have fixed this to keep things consistent for now. Once we start the redesign project we can revisit whether we want to keep this casing or not.

alexandersandberg commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+<div class="link-grid" markdown="1">
+  <ul>
+    <li>
+      <a href="/download/#releases">
+        <div class="flex-container">
+          <div class="latest-release-container">
+          <span>
+            {{ site.data.builds.swift_releases.last.name }}
```
I changed this button to use the latest available patch version instead of a hard-coded minor version and changed it to link to downloads instead of the latest release blog post (because that won't always be available).

IIRC this is what we wanted to do? Let me know if I should revert that.

kaishin commented:
I updated the SVG icons in the landing page to outline versions and fixed the color inversion logic to work in all theme scenarios.

The preview [was updated](https://swift-org-website-kaishin.vercel.app) as well.

alexandersandberg commented on getting-started/swiftui/index.md:
```diff
@@ -0,0 +1,413 @@
+title: Build an iOS app with SwiftUI
+---
+
+> The source code for this guide can be found [on GitHub](https://github.com/0xTim/swift-org-swiftui-tutorial)
```
As discussed in SWWG meeting: we can leave it where it is for now.

0xTim commented:
I've added https://github.com/apple/swift-org-website/pull/373 to update the guides which will need to be merged before this

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+<div class="link-grid" markdown="1">
+  <ul>
+    <li>
+      <a href="/download/#releases">
```
```suggestion
      <a href="/install">
```

tomerd replied:
@kaishin @shahmishal I think it makes more sense to link to the install page instead of the more complicated download page

shahmishal replied:
Sounds good, I accepted the suggestion.

cthielen commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
 ---
 
-# Welcome to Swift.org
+<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> in expert hands. It is <strong>quick</strong> to write, <strong>fast</strong> to run, <strong>modern</strong>, and <strong>safe</strong>.</h1>
```
Let's go with:
`<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> for experts. It is <strong>fast</strong>, <strong>modern</strong>, <strong>safe</strong>, and a <strong>joy</strong> to write.`

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+  </ul>
+</div>
+
+<div class="links links-list-nostyle" markdown="1">
+  - [Learn more about Swift](/about)
+</div>
```
```suggestion
```

tomerd replied:
@daveverwer @dempseyatgithub per @cthielen suggestion I think we can remove this link

cthielen commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
+
+<ul class="use-cases">
+  <li>
+    <h3>Apple platforms</h3>
```
Let's do "Apple Platforms" to keep the casing consistent with "Server and Networking".

tomerd replied:
```suggestion
    <h3>Apple Platforms</h3>
```

tomerd commented on index.md:
```diff
@@ -4,11 +4,130 @@ title: Welcome to Swift.org
 ---
 
-# Welcome to Swift.org
+<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> in expert hands. It is <strong>quick</strong> to write, <strong>fast</strong> to run, <strong>modern</strong>, and <strong>safe</strong>.</h1>
```
```suggestion
<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> for experts. It is <strong>fast</strong>, <strong>modern</strong>, <strong>safe</strong>, and a <strong>joy</strong> to write.
```

tomerd replied:
cc @daveverwer per @cthielen suggestion ^^

daveverwer replied:
I like the change from "in expert hands" to "for experts"

I do prefer "It's quick to write, fast to run, modern, and safe.", but I'm happy if the concensus is otherwise.

tomerd commented on about/index.md:
```diff
@@ -3,15 +3,14 @@ layout: page
 ---
 
-Swift is a general-purpose programming language built using a modern approach to safety, performance, and software design patterns.
+Swift is a general-purpose programming language that's approachable for newcomers and powerful in expert hands. It is quick to write, fast to run, modern, and safe.
```
```suggestion
Swift is a general-purpose programming language that's approachable for newcomers and powerful for experts. It is fast, modern, safe, and a joy to write.
```

tkremenek replied:
I really look this change.

tkremenek replied:
I really like this change.

stevapple commented on install/windows/index.md:
```diff
@@ -110,8 +55,8 @@ Invoke-RestMethod -Url 'get.scoop.sh' | Invoke-Expression
+   
+0. Install Swift:
 
    Swift can be installed through the official installer directly, or using the Windows Package Manager as well.  Notice that Windows Package Manager release may be behind the official release.
```
```suggestion
   Swift can be installed through the official installer directly, or using supported package managers as well.  Notice that latest release from the supported package managers may lag behind the official release.
```

stevapple commented on install/windows/index.md:
```diff
@@ -61,33 +21,18 @@ Windows has the following additional platform specific dependencies:
 
 - Visual Studio (provides the Visual C++ SDK/Build Tools for additional headers)
 
-#### Enabling Developer Mode
-
-In order to develop applications, particularly with the Swift Package Manager, you will need to enable developer mode. Please see Microsoft's [documentation](https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development) for instructions for enabling developer mode.
-
```
Why is this dropped? Developer mode is still required for SwiftPM. It could be mentioned later, but definitely worth the attention.

stevapple reviewed (changes requested):
Still some questions on the Windows installation part.

stevapple commented on install/windows/index.md:
```diff
@@ -61,33 +21,18 @@ Windows has the following additional platform specific dependencies:
-#### Installation Instructions
-
-##### Install using the Windows Package Manager
+## Install using the Windows Package Manager
```
The title should be updated to indicate both WinGet and Scoop. Perhaps a general “Install using package manager”?

In fact I think it may be worth splitting these two because now they’re mostly independent.

shahmishal replied:
@stevapple Would you be able to create a PR with suggested changes?

stevapple replied:
#385

kaishin commented:
@swift-ci test

kaishin commented:
🎊
